parlai/core/image_featurizers.py

```diff
@@ -57,13 +57,35 @@ def init_cnn(self, opt):
             print('[ Using CUDA ]')
             torch.cuda.set_device(opt.get('gpu', -1))
 
-        cnn_type, layer_num = self._image_mode_switcher()
-
-        # initialize the pretrained CNN using pytorch.
-        CNN = getattr(torchvision.models, cnn_type)
-
-        # cut off the additional layer.
-        self.netCNN = nn.Sequential(*list(CNN(pretrained=True).children())[:layer_num])
+        if 'resnet' in self.image_mode:
```

Comment:
resnet vs resnext, so subtle lol.

does it make sense to refactor each of these load processes into some helper methods?

Reply:
yes, I will do so